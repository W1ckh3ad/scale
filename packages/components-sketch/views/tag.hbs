<div data-sketch-artboard="Chips/Tags" class="artboard">
    <h1 class="title--artboard">Tags</h1>
    
    <h2 class="title--artboard-section">Interactive (Clear tag) – Black – Large</h2>
    <div class="agent-states--grid" data-category="Interactive">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-0" dismissable>Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            <scale-tag data-sketch-key="tag-1" dismissable data-sketch-state="hover">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            <scale-tag data-sketch-key="tag-2" dismissable data-sketch-state="focus">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            <scale-tag data-sketch-key="tag-3" dismissable data-sketch-state="active">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            <scale-tag data-sketch-key="tag-4" dismissable disabled="true">Customers</scale-tag>
        </div>
    </div>
    
    <h2 class="title--artboard-section">Interactive (Clear tag) – Black – Small</h2>
    <div class="agent-states--grid" data-category="Interactive">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-5" size="small" dismissable>Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            <scale-tag data-sketch-key="tag-6" size="small" dismissable data-sketch-state="hover">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            <scale-tag data-sketch-key="tag-7" size="small" dismissable data-sketch-state="focus">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            <scale-tag data-sketch-key="tag-8" size="small" dismissable data-sketch-state="active">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            <scale-tag data-sketch-key="tag-9" size="small" dismissable disabled="true">Customers</scale-tag>
        </div>
    </div>
    
    <h2 class="title--artboard-section">Interactive (Clear tag) – Colored – Large</h2>
    <div class="agent-states--grid" data-category="Interactive">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-10" variant="secondary" dismissable>Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            <scale-tag data-sketch-key="tag-11" variant="secondary" dismissable data-sketch-state="hover">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            <scale-tag data-sketch-key="tag-12" variant="secondary" dismissable data-sketch-state="focus">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            <scale-tag data-sketch-key="tag-13" variant="secondary" dismissable data-sketch-state="active">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            <scale-tag data-sketch-key="tag-14" variant="secondary" dismissable disabled="true">Customers</scale-tag>
        </div>
    </div>
    
    <h2 class="title--artboard-section">Interactive (Clear tag) – Colored – Small</h2>
    <div class="agent-states--grid" data-category="Interactive">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-15" variant="secondary" size="small" dismissable>Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            <scale-tag data-sketch-key="tag-16" variant="secondary" size="small" dismissable data-sketch-state="hover">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            <scale-tag data-sketch-key="tag-17" variant="secondary" size="small" dismissable data-sketch-state="focus">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            <scale-tag data-sketch-key="tag-18" variant="secondary" size="small" dismissable data-sketch-state="active">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            <scale-tag data-sketch-key="tag-19" variant="secondary" size="small" dismissable disabled="true">Customers</scale-tag>
        </div>
    </div>

    <h2 class="title--artboard-section">Not Interactive – Black – Large</h2>
    <div class="agent-states--grid" data-category="Static">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-20">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            –
        </div>
    </div>

    <h2 class="title--artboard-section">Not Interactive – Black – Small</h2>
    <div class="agent-states--grid" data-category="Static">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-21" size="small">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            –
        </div>
    </div>

    <h2 class="title--artboard-section">Not Interactive – Colored – Large</h2>
    <div class="agent-states--grid" data-category="Static">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-22" variant="secondary">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            –
        </div>
    </div>

    <h2 class="title--artboard-section">Not Interactive – Colored – Small</h2>
    <div class="agent-states--grid" data-category="Static">
        <div class="agent-states--item">
            <p class="agent-states--label">:enabled</p>
            <scale-tag data-sketch-key="tag-23" variant="secondary" size="small">Customers</scale-tag>
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:hover</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:focus</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:active/:selected</p>
            –
        </div>
        <div class="agent-states--item">
            <p class="agent-states--label">:disabled</p>
            –
        </div>
    </div>
</div>

<script src="/symbol_names.js"></script>
<script>
    Array.from(document.querySelectorAll("scale-tag")).forEach((btn,i) => {
        btn.dataset.sketchSymbol = `Tags / ${
                order(1, getCategoryName(btn))
            } / ${
                order(2, getSizeName(btn))
            } / ${
                order(3, btn.getAttribute('variant') ? 'Black' : 'Colored')
            }${
                getCategoryName(btn) === 'Static'
                ? ''
                : ' / ' + order(4, getStateName(btn))
            }`;
    });
</script>