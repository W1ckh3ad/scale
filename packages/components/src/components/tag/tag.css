.tag {
  --tag-color: var(--scl-color-text-default);
  --tag-text-color: var(--scl-color-white);
  --icon-color: var(--tag-text-color);

  border: 1px solid transparent;
  display: inline-flex;
  outline: none;
  padding: 0 10px;
  text-align: center;
  transition: all 0.15s ease-in-out;
  align-items: center;
  white-space: nowrap;
  border-radius: var(--scl-radius-4);
  vertical-align: baseline;
  justify-content: center;
  font-size: var(--scl-font-variant-body-size);
  line-height: var(--scl-font-variant-body-line-height);
  font-weight: var(--scl-font-weight-bold);
  background: var(--tag-color);
  color: var(--tag-text-color);
  cursor: default;
}

.tag--variant-secondary {
  --tag-color: #00739f;
}

.tag:not(.tag--disabled) scale-icon:hover {
  --icon-color: var(
    --scl-color-primary-active
  ); /* TODO: shouldn't this be hover color? */
}
.tag--dismissable {
  padding-right: 0;
}
.tag--dismissable button {
  border: none;
  cursor: pointer;
  height: 24px;
  margin: 0;
  outline: none;
  padding: 0;
  background: transparent;
  margin-left: var(--scl-spacing-2);
  border-radius: var(--scl-radius-4);
}

.tag--dismissable scale-icon svg {
  outline: none;
}
.tag--dismissable button:focus {
  border: 1px solid var(--scl-color-background);
  background: var(--scl-color-black);
  box-shadow:  0 0 0 var(--scl-spacing-2) var(--scl-color-focus);
}
.tag--dismissable button:hover {
  background: var(--scl-color-black);
}
.tag--dismissable button:focus scale-icon {
  top: -1px;
  position: relative;
}
.tag--size-small {
  padding: 0 var(--scl-spacing-8);
  font-size: var(--scl-font-size-12);
  line-height: var(--scl-font-line-height-160);
}
.tag--size-small.tag--dismissable {
  padding-right: 0;
}
.tag--size-small.tag--dismissable button {
  height: 20px;
}

.tag--variant-secondary.tag--disabled {
  opacity: 0.5;
}
.tag--link {
  text-decoration: none;
}
.tag--link:focus {
  border: 1px solid white;
  box-shadow:  0 0 0 var(--scl-spacing-2) var(--scl-color-focus);
}
.tag--link.tag--variant-secondary {
  background: var(--tag-color);
}
.tag--link.tag--variant-secondary:hover {
  background: var(--tag-color);
}
.tag--link.tag--variant-secondary:focus {
  border: 1px solid white;
}
.tag--disabled {
  --tag-color: var(--scl-color-background-disabled);
  --tag-text-color: var(--scl-color-grey-60);

  cursor: not-allowed;
}
.tag--disabled button {
  cursor: not-allowed;
}
.tag--disabled button:hover {
  cursor: not-allowed;
}
