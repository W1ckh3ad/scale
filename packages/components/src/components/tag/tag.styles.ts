import { JssStyle } from 'jss';

export const styles: JssStyle = {
  tag: {
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: 16,
    fontWeight: 400,
    lineHeight: '24px',
    textAlign: 'center',
    whiteSpace: 'nowrap',
    verticalAlign: 'baseline',
    borderRadius: 4,
    transition: 'all 0.15s ease-in-out',
    background: ({ palette }) => palette.gray,
    color: ({ palette }) => palette.white,
    '--icon-color': ({ palette }) => palette.white,
    padding: '0px 10px',
    fontFamily: 'TeleNeoWeb-Bold',
    '& scale-icon': {
      marginLeft: 2,
    },
    '&:not($tag--disabled)': {
      '& scale-icon': {
        '&:hover': {
          '--icon-color': ({ palette }) => palette.magentaActive,
        },
      },
    },
  },
  'tag--dismissable': {
    paddingRight: 2,
  },
  'tag--size-small': {
    padding: '4px 8px',
    fontSize: 12,
    '& scale-icon': {
      height: 12,
      width: 12,
    },
    '&$tag--dismissable': {
      paddingRight: 4,
    },
  },
  'tag--variant-secondary': {
    background: ({ palette }) => palette.secondary,
    border: ({ palette }) => `1px solid ${palette.secondary}`,
    '&$tag--disabled': {
      background: ({ palette }) => palette.secondary,
      color: ({ palette }) => palette.white,
      '--icon-color': ({ palette }) => palette.white,
      border: 'none',
      opacity: '0.5',
    },
  },
  'tag--link': {
    textDecoration: 'none',
    '&$tag--variant-secondary': {
      background: ({ palette }) => palette.secondary,
      border: ({ palette }) => `1px solid ${palette.secondary}`,
      '&:hover': {
        background: ({ palette }) => palette.secondary,
        border: ({ palette }) => `1px solid ${palette.secondary}`,
      },
      '&:focus': {
        background: ({ palette }) => palette.secondary,
        border: ({ palette }) => `1px solid ${palette.secondary}`,
      },
    },
  },
  'tag--disabled': {
    color: ({ palette }) => palette.grayDisabledText,
    background: ({ palette }) => palette.grayDisabledBackground,
    '--icon-color': ({ palette }) => palette.grayDisabledText,
    cursor: 'not-allowed',
  },
};
