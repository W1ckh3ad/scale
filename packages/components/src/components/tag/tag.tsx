import { Component, Prop, h, Host, Event, EventEmitter } from '@stencil/core';
import classNames from 'classnames';
import Base from '../../utils/base-interface';

const name = 'tag';
@Component({
  tag: 'scale-tag',
  styleUrl: './tag.css',
  shadow: true,
})
export class Tag implements Base {
  /** (optional) Tag size */
  @Prop() size?: string = '';
  /** (optional) Tag variant */
  @Prop() variant?: string = '';
  /** (optional) Tag href */
  @Prop() href?: string = '';
  /** (optional) Tag target */
  @Prop() target?: string = '_self';
  /** (optional) Tag dismissable */
  @Prop() dismissable?: boolean = false;
  /** (optional) Tag disabled */
  @Prop() disabled?: boolean = false;
  /** (optional) Dismiss label */
  @Prop() dismissText?: string = 'dismiss';
  /** (optional) Injected CSS styles */
  @Prop() styles?: string;

  /** (optional) Close icon click event */
  @Event() scaleClose: EventEmitter<MouseEvent>;

  componentWillUpdate() {}
  disconnectedCallback() {}

  handleClose(event) {
    event.preventDefault();
    event.stopPropagation();
    this.scaleClose.emit(event);
  }

  render() {
    const Element = !!this.href && !this.disabled ? 'a' : 'span';
    const linkProps = !!this.href
      ? {
          href: this.href,
          target: this.target,
        }
      : {};
    const iconProps = !this.disabled
      ? {
          onClick: event => this.handleClose(event),
        }
      : {};

    return (
      <Host>
        {this.styles && <style>{this.styles}</style>}

        <Element class={this.getCssClassMap()} {...linkProps}>
          <slot />

          {this.dismissable && (
            <button disabled={this.disabled} aria-label={this.dismissText}>
              <scale-icon
                size={this.size === 'small' ? 20 : 24}
                path="M13.77 12l4.415-4.42a1.25 1.25 0 10-1.765-1.765L12 10.23 7.58 5.815A1.25 1.25 0 105.815 7.58L10.23 12l-4.415 4.42a1.245 1.245 0 000 1.765c.232.237.549.37.88.37.332 0 .65-.134.885-.37L12 13.77l4.42 4.415c.234.236.553.37.885.37.331 0 .648-.133.88-.37a1.245 1.245 0 000-1.765L13.77 12z"
                {...iconProps}
              />
            </button>
          )}
        </Element>
      </Host>
    );
  }

  getCssClassMap() {
    return classNames(
      name,
      this.size && `${name}--size-${this.size}`,
      this.variant && `${name}--variant-${this.variant}`,
      !!this.href && `${name}--link`,
      !!this.dismissable && `${name}--dismissable`,
      !!this.disabled && `${name}--disabled`
    );
  }
}
