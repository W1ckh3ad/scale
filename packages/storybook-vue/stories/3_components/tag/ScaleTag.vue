<template>
  <scale-tag
    :disabled="disabled"
    :dismissable="dismissable"
    :dismiss-text="dismissText"
    :href="href"
    :size="size"
    :target="target"
    :variant="variant"
    @click="click"
    @scaleClose="click"
    :styles="styles"
  >
    <slot></slot>
  </scale-tag>
</template>

<script>
import { action } from "@storybook/addon-actions";

export default {
  props: {
    disabled: { type: Boolean, default: false },
    dismissable: { type: Boolean, default: false },
    href: String,
    dismissText: String,
    size: String,
    target: { type: String, default: "_self" },
    variant: String,
    styles: String,
  },
  methods: {
    click: function ($event) {
      action("click");
      this.$emit("click", $event);
    },
  },
};
</script>
