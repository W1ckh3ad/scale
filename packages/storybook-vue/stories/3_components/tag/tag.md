# Chip

Chips help users to percieve and identify tags and categories assigned to objects in the product. By default, chips are not interactive.

## General

Chips allow users to enter information, make selections or filter content. Chips should appear dynamically as a group of multiple interactive elements.

### When to Use

Use chips when content is mapped to multiple categories and the user needs a way to differentiate between them. Chips can also be used as a method of filtering data, to show only items within a particular category.

## Basic Types

### Static

For display only.

![Image Name](/assets/3_components/chip/image-20200810103235995.png)

### Interactive

Can be removed directly.

![Image Name](/assets/3_components/chip/image-20200810103222682.png)

## Colors

Use a default color or assign a custom color. When customizing, use a color-coding scheme that is clear and supports the user in what they want to achieve.

### Default

![Image Name](/assets/3_components/chip/image-20200810103222682.png)
### Custom

Consider accessibility / contrast in your color choice.

![Image Name](/assets/3_components/chip/image-20200810103229377.png)

## Sizes

Regular is the recommended size.

### Regular

![Image Name](/assets/3_components/chip/image-20200810103222682.png)

### Small

Use the small size only in exceptional cases when space is limited.

![Image Name](/assets/3_components/chip/image-20200810103241914.png)

## Elements

### Label

The label clearly describes what the chip stands for. Use a maximum of three words.

### Remove Icon (Optional)

When users can manually remove tags or categories, an X - icon can be added to the chip. Clicking on it removes the tag from its assigned object.

## Related Components

Button, Toggle

## Code
