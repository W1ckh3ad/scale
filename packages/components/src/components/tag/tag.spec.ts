import { newSpecPage } from '@stencil/core/testing';
import { Tag } from './tag';

describe('Tag', () => {
  let element;
  beforeEach(async () => {
    element = new Tag();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Tag],
      html: `<scale-tag>Label</scale-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have a link', async () => {
    const page = await newSpecPage({
      components: [Tag],
      html: `<scale-tag href="#">Label</scale-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.size = 'small';
    expect(element.getCssClassMap()).toContain('tag--size-small');

    element.variant = 'primary';
    expect(element.getCssClassMap()).toContain('tag--variant-primary');

    element.href = 'http://example.com';
    expect(element.getCssClassMap()).toContain('tag--link');
  });
});
