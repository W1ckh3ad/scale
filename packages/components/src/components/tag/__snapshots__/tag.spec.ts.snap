// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should have a link 1`] = `
<scale-tag href="#">
  <mock:shadow-root>
    <style>
      .tag {
  border: 1px solid transparent;
  display: inline-flex;
  outline: none;
  padding: 0 10px;
  text-align: center;
  transition: all 0.15s ease-in-out;
  align-items: center;
  white-space: nowrap;
  border-radius: 4px;
  vertical-align: baseline;
  justify-content: center;
  font-size: 1rem;
  line-height: 1.5;
  background: #262626;
  color: #FFF;
  --icon-color: #FFF;
  font-weight: 700;
}
.tag:not(.tag--disabled) {}
.tag:not(.tag--disabled) scale-icon {}
.tag:not(.tag--disabled) scale-icon:hover {
  --icon-color: #CB0068;
}
.tag--dismissable {
  padding-right: 0;
}
.tag--dismissable button {
  border: none;
  cursor: pointer;
  height: 24px;
  margin: 0;
  outline: none;
  padding: 0;
  background: transparent;
  margin-left: 2px;
  border-radius: 4px;
}
.tag--dismissable scale-icon {}
.tag--dismissable scale-icon svg {
  outline: none;
}
.tag--dismissable button:focus {
  border: 1px solid white;
  background: black;
  box-shadow: 0 0 0 2px #009DE0;
}
.tag--dismissable button:hover {
  background: black;
}
.tag--dismissable button:focus scale-icon {
  top: -1px;
  position: relative;
}
.tag--size-small {
  padding: 0 8px;
  font-size: 0.75rem;
  line-height: 1.6;
}
.tag--size-small.tag--dismissable {
  padding-right: 0;
}
.tag--size-small.tag--dismissable button {
  height: 20px;
}
.tag--variant-secondary {
  background: #00739F;
}
.tag--variant-secondary.tag--disabled {
  opacity: 0.5;
  background: #00739F;
  color: #FFF;
  --icon-color: #FFF;
}
.tag--link {
  text-decoration: none;
}
.tag--link:focus {
  border: 1px solid white;
  box-shadow: 0 0 0 2px #009DE0;
}
.tag--link.tag--variant-secondary {
  background: #00739F;
}
.tag--link.tag--variant-secondary:hover {
  background: #00739F;
}
.tag--link.tag--variant-secondary:focus {
  border: 1px solid white;
  background: #00739F;
}
.tag--disabled {
  color: #A4A4A4;
  cursor: not-allowed;
  --icon-color: #A4A4A4;
  background: #D0D0D0;
}
.tag--disabled button {
  cursor: not-allowed;
  background: #D0D0D0;
}
.tag--disabled button:hover {
  cursor: not-allowed;
  background: #D0D0D0;
}
    </style>
    <a class="tag--link tag" href="#" target="_self">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</scale-tag>
`;

exports[`Tag should match snapshot 1`] = `
<scale-tag>
  <mock:shadow-root>
    <style>
      .tag {
  border: 1px solid transparent;
  display: inline-flex;
  outline: none;
  padding: 0 10px;
  text-align: center;
  transition: all 0.15s ease-in-out;
  align-items: center;
  white-space: nowrap;
  border-radius: 4px;
  vertical-align: baseline;
  justify-content: center;
  font-size: 1rem;
  line-height: 1.5;
  background: #262626;
  color: #FFF;
  --icon-color: #FFF;
  font-weight: 700;
}
.tag:not(.tag--disabled) {}
.tag:not(.tag--disabled) scale-icon {}
.tag:not(.tag--disabled) scale-icon:hover {
  --icon-color: #CB0068;
}
.tag--dismissable {
  padding-right: 0;
}
.tag--dismissable button {
  border: none;
  cursor: pointer;
  height: 24px;
  margin: 0;
  outline: none;
  padding: 0;
  background: transparent;
  margin-left: 2px;
  border-radius: 4px;
}
.tag--dismissable scale-icon {}
.tag--dismissable scale-icon svg {
  outline: none;
}
.tag--dismissable button:focus {
  border: 1px solid white;
  background: black;
  box-shadow: 0 0 0 2px #009DE0;
}
.tag--dismissable button:hover {
  background: black;
}
.tag--dismissable button:focus scale-icon {
  top: -1px;
  position: relative;
}
.tag--size-small {
  padding: 0 8px;
  font-size: 0.75rem;
  line-height: 1.6;
}
.tag--size-small.tag--dismissable {
  padding-right: 0;
}
.tag--size-small.tag--dismissable button {
  height: 20px;
}
.tag--variant-secondary {
  background: #00739F;
}
.tag--variant-secondary.tag--disabled {
  opacity: 0.5;
  background: #00739F;
  color: #FFF;
  --icon-color: #FFF;
}
.tag--link {
  text-decoration: none;
}
.tag--link:focus {
  border: 1px solid white;
  box-shadow: 0 0 0 2px #009DE0;
}
.tag--link.tag--variant-secondary {
  background: #00739F;
}
.tag--link.tag--variant-secondary:hover {
  background: #00739F;
}
.tag--link.tag--variant-secondary:focus {
  border: 1px solid white;
  background: #00739F;
}
.tag--disabled {
  color: #A4A4A4;
  cursor: not-allowed;
  --icon-color: #A4A4A4;
  background: #D0D0D0;
}
.tag--disabled button {
  cursor: not-allowed;
  background: #D0D0D0;
}
.tag--disabled button:hover {
  cursor: not-allowed;
  background: #D0D0D0;
}
    </style>
    <span class="tag">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Label
</scale-tag>
`;
